File: GeneratedEntityReaderClasses/Components/CutThroughWorldDoneHereComponent.cs
namespace GeneratedNoitaClasses;
public class CutThroughWorldDoneHereComponent : NoitaComponentBase,  iNoitaType<CutThroughWorldDoneHereComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaUInt id_of_done_cut { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		id_of_done_cut.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		id_of_done_cut.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/CutThroughWorldDoneHereComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUInt

from NoitaStream import NoitaStream
@dataclass
class CutThroughWorldDoneHereComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    id_of_done_cut:NoitaUInt = field(default_factory=NoitaUInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.id_of_done_cut.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.id_of_done_cut.Write(s)

